File: 0648-replace-words/0648-replace-words.cpp
class Solution {
public:
    string replaceWords(vector<string>& dictionary, string sentence) {
        sort(dictionary.begin(), dictionary.end());
        string answer = "";
        
        istringstream ss(sentence);
        string buffer;
        
        int count;
        while(getline(ss, buffer, ' ')) {
            for(string d : dictionary) {
                count = 0;
                for(int i = 0; i < d.size(); i++) {
                    if(d[i] != buffer[i]) break;
                    count++;
                }
                
                if(count == d.size()) {
                    buffer = d;
                    break;
                }
            }
            
            answer += buffer + " ";
        }
        
        answer.pop_back();
        return answer;
    }
};

File: 0648-replace-words/0648-replace-words.py
class Solution:
    def replaceWords(self, dictionary: List[str], sentence: str) -> str:
        answer = list()
        dictionary.sort()
        
        for s in sentence.split():
            for i in dictionary:
                if i in s[0:len(i)]:
                    s = i
                    break
            answer.append(s)
        return ' '.join(answer)
        